contracts/DADBridgeWrapper.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: SEE LICENSE IN LICENSE
+pragma solidity 0.8.9;
+
+import { AccessControl } from "@openzeppelin/contracts/access/AccessControl.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract DADBridgeWrapper is AccessControl {
+    using SafeERC20 for IERC20;
+
+    bytes32 public constant ROLE_ADMIN = keccak256("ROLE_ADMIN");
+
+    IERC20 public immutable bntToken;
+    IERC20 public immutable ethBntToken;
+
+    uint256 public totalSupply;
+
+    constructor(IERC20 _bntToken, IERC20 _ethBntToken) {
+        bntToken = _bntToken;
+        ethBntToken = _ethBntToken;
+        _setRoleAdmin(ROLE_ADMIN, ROLE_ADMIN);
+        _setupRole(ROLE_ADMIN, msg.sender);
+    }
+
+    function mint(address recipient, uint256 amount) external onlyRole(ROLE_ADMIN) {
+        require(recipient != address(0), "ERR_INVALID_RECIPIENT");
+        require(amount > 0, "ERR_INVALID_AMOUNT");
+        bntToken.safeTransfer(recipient, amount);
+        ethBntToken.safeTransfer(recipient, amount);
+        totalSupply += amount;
+    }
+
+    function burn(address recipient, uint256 amount) external pure {
```

Comment:
TODO:
In order to avoid this function showing up on Etherscan/ReadContract, remove the `pure` and "live with the compilation warning".

Reply:
no problem having this showing up on etherscan

Reply:
NP, but keep in mind the following:
1. Executing it within an offchain transaction - well, I'm not quite sure how it works with constant functions which don't require mining etc.
2. Executing it onchain (i.e., from another contract function) - if the caller doesn't do any "other" state-changing operation, then they will get the same warning we would get if we removed the `pure`.

Reply:
it's just not going to work but that's fine

Reply:
Why does it needs to be `pure`? To eliminate the warning?

Reply:
Yes.
I can add `this;` in order to reduce the warning from `pure` to `view`.
But I have no idea how to get rid of this warning completely (other than actually doing some state-changing operation, like emitting a dummy event or whatever).

Reply:
BTW, we also don't ignore state variables this way, but rather comment them out (for example, see StakingRewards.sol `onAddingLiquidity`)

Reply:
There's a newer trick to suppress this warning: you can add `virtual` and then the warning will be suppressed. IMHO, let's do it.

Reply:
Done.